Sources/WhisperKit/Core/LogitsFilter.swift

```diff
@@ -46,19 +47,158 @@ public class SuppressBlankFilter: LogitsFiltering {
     }
 }
 
+/// Implementation based on https://github.com/openai/whisper/blob/master/whisper/decoding.py#L441
 @available(macOS 14, iOS 17, watchOS 10, visionOS 1, *)
 public class TimestampRulesFilter: LogitsFiltering {
-    let tokenizer: Tokenizer
+    let noTimestampsToken: Int
+    let timeTokenBegin: Int
+    let endToken: Int
     let sampleBegin: Int
-    let maxInitialTimestamp: Int?
+    let maxInitialTimestampIndex: Int?
 
-    public init(tokenizer: Tokenizer, sampleBegin: Int) {
-        // TODO: implement
-        fatalError("Not implemented: \(#function)")
+    public init(
+        noTimestampsToken: Int,
+        timeTokenBegin: Int,
+        endToken: Int,
+        sampleBegin: Int,
+        maxInitialTimestampIndex: Int?
+    ) {
+        self.noTimestampsToken = noTimestampsToken
+        self.timeTokenBegin = timeTokenBegin
+        self.endToken = endToken
+        self.sampleBegin = 3 // FIXME: it should not be hardcoded value
```

Comment:
this needs to be fixed before merging

Reply:
Yes, this breaks the EN models I believe